### PR TITLE
externalize script

### PR DESCRIPTION
**Pull request by aknguyen7**

### CODE REVIEW

File: js/interactive-guides/externalized-string.js [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    return translateMsg;
+}
+
+function appendDate(element, translateMsg) {
+    var date = translateMsg + element.innerHTML;
+    return date;
+}
+
+/**
+ * Query for all data-externailzedString and replace the string for translate messages into key=value array
```
Comment (kevinwlau) [inline js/interactive-guides/externalized-string.js]: typo on `data-externailzedString`

File: js/interactive-guides/externalized-string.js [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+/**
+ * Query all the data-externalizedString in html to translate the message.
+ */
+function retrieveExternalizedStrings(rootDir) {
+    'use strict';
+        
+    var languageCode = getLanguageCode();
+
+    var nlsFile = "messages.js";
+    var url = (languageCode !== "en") ? rootDir + languageCode + "/" + nlsFile :  rootDir + nlsFile;
```
Comment (kevinwlau) [inline js/interactive-guides/externalized-string.js]: This is assuming `rootDir` ends with a `/`. Maybe note it in comment or have a check for it.

File: js/interactive-guides/externalized-string.js [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+    
+    var httpReq = new XMLHttpRequest();
+    var params = null;
+    
+    httpReq.open("GET", url, true);
+
+    // Call a function when the state changes.
+    httpReq.onreadystatechange = function () {
+        if (httpReq.readyState === 4 && httpReq.status === 200) {   
+            replaceExternalizedStrings(httpReq.responseText);
```
Comment (kevinwlau) [inline js/interactive-guides/externalized-string.js]: Does this call to `replaceExternalizedStrings` require the `httpSendRequest` to have it passed into the parameters?

Comment (aknguyen7) [reply js/interactive-guides/externalized-string.js]: no, it doesn't have to pass into as parameter, I removed it

**Pull request merged**